iron-input.html

```diff
@@ -125,11 +169,52 @@
      * @suppress {checkTypes}
      */
     _bindValueChanged: function() {
+      if (typeof this.bindValue !== 'string') {
+        // it will trigger another _bindValueChanged event
+        this.bindValue = !(this.bindValue || this.bindValue === 0) ? '' : this.bindValue + '';
+      } else {
+        this._updateValueIfNeeded();
+        // manually notify because we don't want to notify until after setting value
+        this.fire('bind-value-changed', {value: this.bindValue});
+      }
+    },
+
+    _bindValueAsNumberChanged: function() {
+      // ensure it is a number
+      if (typeof this.bindValueAsNumber !== 'number') {
+        // it will trigger another _bindValueAsNumberChanged event
+        if (isNaN(this.bindValueAsNumber)) {
```

Comment:
If you already checked that `typeof this.bindValueAsNumber !== 'number', it seems as though `isNaN(this.bindValueAsNumber)` would be implicitly false (since `typeof NaN === 'number'`). Is there a side-effect of the first conditional check that I should be considering?